bots/tests-invoke

```diff
@@ -235,6 +246,20 @@ class PullTask(object):
         sys.stderr.write(msg)
 
         ret = None
+
+        offline = ('offline' in opts) and opts.offline
+        if not ret and self.base and "TEST_INVOKE_RESPAWN" not in os.environ:
+            ret = ret or self.rebase(offline)
+
+        # Did we rebase, if so rexec
+        if not ret and self.respawn:
+            os.environ["TEST_INVOKE_RESPAWN"] = "1"
```

Comment:
If this is being run with `TEST_INVOKE_RESPAWN=1`, then `ret` is guaranteed to be `None`, so we rely  on `self.respawn` being `False` to avoid an endless respawn loop. Now, this Should not Happen™ as we already rebased, but  it would be good to add an assertion above: `assert not (TEST_INVOKE_RESPAWN" in os.environ and self.respawn)`, which will make debugging much easier.